test/python/primitives/containers/test_sampler_pub.py

```diff
@@ -0,0 +1,329 @@
+
+    @ddt.idata(range(5))
+    def test_validate_num_parameters(self, num_params):
+        """Test unparameterized circuit raises for parameter values"""
+        params = (Parameter("a"), Parameter("b"))
+        circuit = QuantumCircuit(2)
+        circuit.rx(params[0], 0)
+        circuit.ry(params[1], 1)
+        circuit.measure_all()
+        parameter_values = BindingsArray(np.zeros((2, num_params)), shape=2)
+        if num_params == len(params):
+            SamplerPub(circuit, parameter_values=parameter_values)
+            return
+
+        with self.assertRaises(ValueError):
+            SamplerPub(circuit, parameter_values=parameter_values)
+
+    @ddt.data((), (3,), (2, 3))
+    def test_shaped_zero_parameter_values(self, shape):
+        """Test Passing in a shaped array with no parameters works"""
+        circuit = QuantumCircuit(2)
+        parameter_values = BindingsArray(np.zeros((*shape, 0)), shape=shape)
+        pub = SamplerPub(circuit, parameter_values=parameter_values)
+        self.assertEqual(pub.shape, shape)
+
+    def test_coerce_circuit(self):
+        """Test coercing an unparameterized circuit"""
+        circuit = QuantumCircuit(10)
+        circuit.measure_all()
+
+        pub = SamplerPub.coerce(circuit)
+        self.assertEqual(pub.circuit, circuit, msg="incorrect value for `circuit` property")
+        self.assertEqual(pub.shots, None, msg="incorrect value for `shots` property")
+        # Check bindings array, this is more cumbersome since the class doesn't have an eq method
+        self.assertIsInstance(
+            pub.parameter_values,
+            BindingsArray,
+            msg="incorrect type for `parameter_values` property",
+        )
+        self.assertEqual(
+            pub.parameter_values.shape, (), msg="incorrect shape for `parameter_values` property"
+        )
+        self.assertEqual(
+            pub.parameter_values.num_parameters,
+            0,
+            msg="incorrect num parameters for `parameter_values` property",
+        )
+
+    def test_invalid_coerce_circuit(self):
+        """Test coercing parameterized circuit raises"""
+        params = (Parameter("a"), Parameter("b"))
+        circuit = QuantumCircuit(10)
+        circuit.rx(params[0], 0)
+        circuit.ry(params[1], 1)
+        circuit.measure_all()
+        with self.assertRaises(ValueError):
+            SamplerPub.coerce(circuit)
+
+    @ddt.data(1, 10, 100, 1000)
+    def test_coerce_pub_with_shots(self, shots):
+        """Test coercing a SamplerPub"""
+        params = (Parameter("a"), Parameter("b"))
+        circuit = QuantumCircuit(2)
+        circuit.rx(params[0], 0)
+        circuit.ry(params[1], 1)
+        circuit.measure_all()
+        pub1 = SamplerPub(
+            circuit=circuit,
+            parameter_values=BindingsArray(kwvals={params: np.ones((10, 2))}),
+            shots=1000,
+        )
+        pub2 = SamplerPub.coerce(pub1, shots=shots)
+        self.assertEqual(pub1, pub2)
```

Comment:
How is this passing? It looks like `__eq__` has not been implemented

Reply:
In absence of eq method it functions like `pub1 is pub2`